core/domain/state_domain.py

```diff
@@ -999,25 +1332,34 @@ def validate(self):
 class WrittenTranslation(python_utils.OBJECT):
     """Value object representing a written translation for a content."""
 
-    def __init__(self, html, needs_update):
+    def __init__(self, data_format, translation, needs_update):
         """Initializes a WrittenTranslation domain object.
 
         Args:
-            html: str. A piece of user submitted HTML. This is cleaned in such
-                a way as to contain a restricted set of HTML tags.
-            needs_update: bool. Whether html is marked for needing review.
+            data_format: str. 'unicode' or 'html'. Indicates if translation
+                field is html or unicode.
+            translation: str. A piece of user submitted HTML or unicode. If
+                html, this is cleaned in such a way as to contain a restricted
+                set of HTML tags.
+            needs_update: bool. Whether translation is marked as needing
+                review.
         """
-        self.html = html_cleaner.clean(html)
+        self.data_format = data_format
+        self.translation = translation
         self.needs_update = needs_update
 
+        if data_format == 'html':
```

Comment:
DATA_FORMAT_HTML (especially if you're using this string anywhere else). 

Please fix elsewhere too -- don't leave hardcoded strings in the codebase.